tensorflow_addons/activations/gelu_test.py

```diff
@@ -19,58 +19,33 @@
 
 from absl.testing import parameterized
 
-import math
-
 import numpy as np
 import tensorflow as tf
 from tensorflow_addons.activations import gelu
 from tensorflow_addons.utils import test_utils
 
 
-def _ref_gelu(x, approximate=True):
-    x = tf.convert_to_tensor(x)
-    if approximate:
-        pi = tf.cast(math.pi, x.dtype)
-        coeff = tf.cast(0.044715, x.dtype)
-        return 0.5 * x * (
-            1.0 + tf.tanh(tf.sqrt(2.0 / pi) * (x + coeff * tf.pow(x, 3))))
-    else:
-        return 0.5 * x * (
-            1.0 + tf.math.erf(x / tf.cast(tf.sqrt(2.0), x.dtype)))
-
-
 @test_utils.run_all_in_graph_and_eager_modes
 class GeluTest(tf.test.TestCase, parameterized.TestCase):
     @parameterized.named_parameters(("float16", np.float16),
                                     ("float32", np.float32),
                                     ("float64", np.float64))
     def test_gelu(self, dtype):
-        x = np.random.rand(2, 3, 4).astype(dtype)
-        self.assertAllCloseAccordingToType(gelu(x), _ref_gelu(x))
-        self.assertAllCloseAccordingToType(gelu(x, False), _ref_gelu(x, False))
+        x = tf.constant([-2.0, -1.0, 0.0, 1.0, 2.0], dtype=dtype)
+        expected_result = tf.constant(
+            [-0.04540229, -0.158808, 0.0, 0.841192, 1.9545977], dtype=dtype)
+        self.assertAllCloseAccordingToType(gelu(x), expected_result)
 
-    @parameterized.named_parameters(("float16", np.float16),
-                                    ("float32", np.float32),
-                                    ("float64", np.float64))
-    def test_gradients(self, dtype):
-        x = tf.constant([1.0, 2.0, 3.0], dtype=dtype)
-
-        for approximate in [True, False]:
-            with self.subTest(approximate=approximate):
-                with tf.GradientTape(persistent=True) as tape:
-                    tape.watch(x)
-                    y_ref = _ref_gelu(x, approximate)
-                    y = gelu(x, approximate)
-                grad_ref = tape.gradient(y_ref, x)
-                grad = tape.gradient(y, x)
-                self.assertAllCloseAccordingToType(grad, grad_ref)
+        expected_result = tf.constant(
+            [-0.04550028, -0.15865526, 0.0, 0.8413447, 1.9544997], dtype=dtype)
+        self.assertAllCloseAccordingToType(gelu(x, False), expected_result)
 
     @parameterized.named_parameters(("float32", np.float32),
                                     ("float64", np.float64))
     def test_theoretical_gradients(self, dtype):
         # Only test theoretical gradients for float32 and float64
         # because of the instability of float16 while computing jacobian
-        x = tf.constant([1.0, 2.0, 3.0], dtype=dtype)
+        x = tf.constant([-2.0, -1.0, 0.0, 1.0, 2.0], dtype=dtype)
```

Comment:
Should it be in the README that we test gradients across positive/zero/negative or should we leave that to be specific to activation function

Reply:
Probably want to test some special values like 0 or some edge cases, so I think it's better to leave it as activation-specific test.